File: 練習問題/1-5/問題2：線形探索（名前探し）.ts
const isTargetExist: (items: number[], target: number) => boolean = (items, target) => {
    for (let i = 0; i < items.length; i++) {
        if (items[i] === target) {
            return true;
        }
    }
    return false;
};

const targetInt: number = 3;
const checkItems1: number[] = [1, 2, 3, 4, 5];
const checkItems2: number[] = [1, 2, 4, 5];

console.log(isTargetExist(checkItems1, targetInt));
console.log(isTargetExist(checkItems2, targetInt));

File: 練習問題/1-5/問題2：線形探索（名前探し）.c
# include <stdio.h>
# include <stdbool.h>

int main(void) {
  int target = 3;
  int checkItems1[] = {1,2,3,4,5};
  int checkItems2[] = {1,2,3,4};
  bool result = false;

  int length1 = sizeof(checkItems1) / sizeof(checkItems1[0]);
  int length2 = sizeof(checkItems2) / sizeof(checkItems2[0]);

  for (int i = 0; i < length1; i++) {
    if (checkItems1[i] == target) {
      result = true;
      break;
    }
  }

  printf("%s\n", result ? "true" : "false");
  return 0;
}